### PR TITLE
Add SonarCloud Skills Guide, expand README examples, and route vulnerabilities to outlaw-tracker

### PR DESCRIPTION
**Pull request by jordanpadams**
## 🗒️ Summary

Two improvements to the SonarCloud skills workflow:

1. **SonarCloud Skills Guide** (`docs/SONARCLOUD_SKILLS_GUIDE.md`) — new standalone setup and usage guide covering prerequisites, example prompts, the full export→triage→update pipeline, and troubleshooting tips.

2. **README usage examples** — expanded the Usage section with sonarcloud-skills pipeline examples and links to the new guide.

3. **Route security vulnerabilities to `NASA-PDS/outlaw-tracker`** — the `sonarcloud-security-triaging` skill now creates GitHub issues in the private `NASA-PDS/outlaw-tracker` repo for any confirmed true positive vulnerabilities, keeping security tracking self-contained in a private space.

### 🤖 AI Assistance Disclosure
- [ ] No AI assistance used
- [ ] AI used for light assistance (e.g., suggestions, refactoring, documentation help, minor edits)
- [x] AI used for moderate content generation (AI generated some code or logic, but the developer authored or heavily revised the majority)
- [ ] AI generated substantial portions of this code

Estimated % of code influenced by AI: 80%

## ⚙️ Test Data and/or Report

Documentation and skill instruction changes only — no executable code modified. Changes were validated by reviewing the updated SKILL.md sections for correctness and completeness.

## ♻️ Related Issues

N/A

## 🤓 Reviewer Checklist

*Reviewers: Please verify the following before approving this pull request.*

### Documentation and PR Content
- [ ] **Documentation:** README, Wiki, or inline documentation (Sphinx, Javadoc, Docstrings) have been updated to reflect these changes.
- [ ] **Issue Traceability:** The PR is linked to a valid GitHub Issue
- [ ] **PR Title:** The PR title is "user-friendly" clearly identifying what is being fixed or the new feature being added, that if you saw it in the Release Notes for a tool, you would be able to get the gist of what was done.

### Security & Quality
- [ ] **SonarCloud:** Confirmed no new High or Critical security findings.
- [ ] **Secrets Detection:** Verified that the Secrets Detection scan passed and no sensitive information (keys, tokens, PII) is exposed.
- [ ] **Code Quality:** Code follows organization style guidelines and best practices for the specific language (e.g., PEP 8, Google Java Style).

### Testing & Validation
- [ ] **Test Accuracy:** Verified that test data is accurate, representative of real-world PDS4 scenarios, and sufficient for the logic being tested.
- [ ] **Coverage:** Automated tests cover new logic and edge cases.
- [ ] **Local Verification:** (If applicable) Successfully built and ran the changes in a local or staging environment.

### Maintenance
- [ ] **Backward Compatibility:** Confirmed that these changes do not break existing downstream dependencies or API contracts (or that breaking changes are clearly documented).

---
🤖 Generated with assistance from [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: static/marketplace/skills/dependabot-alerts-exporting/scripts/fetch-dependabot-alerts.mjs [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+  return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+async function fetchWithRetry(url, options = {}, retries = 3) {
+  for (let attempt = 1; attempt <= retries; attempt++) {
+    try {
+      const response = await fetch(url, { ...options, headers });
+
+      // Rate limiting
+      if (response.status === 429 || response.status === 403) {
```
Comment (nutjob4life) [inline static/marketplace/skills/dependabot-alerts-exporting/scripts/fetch-dependabot-alerts.mjs]: AI claims that: 403 has multiple meanings in GitHub APIs, including rate limit, permission denied, and repo access issues, and suggests narrowing this down with:
```javascript
if (response.status === 403 && response.headers.get('x-ratelimit-remaining') === '0')
```

File: static/marketplace/skills/dependabot-alerts-exporting/scripts/fetch-dependabot-alerts.mjs [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+    alerts: allAlerts
+  };
+
+  writeFileSync(outputFile, JSON.stringify(output, null, 2));
+
+  console.log('\n✅ Export complete!');
+  console.log(`   Output: ${outputFile}`);
+  console.log(`   Total alerts: ${allAlerts.length}`);
+  console.log('   By severity:');
+  for (const [sev, count] of Object.entries(output.summary.bySeverity).sort()) {
```
Comment (nutjob4life) [inline static/marketplace/skills/dependabot-alerts-exporting/scripts/fetch-dependabot-alerts.mjs]: The `.sort()` here puts the order as HIGH, LOW, MEDIUM which is confusing.

Maybe define the severity explicity: 
```javascript
const severityOrder = ['critical', 'high', 'medium', 'low'];
```

**Review by nutjob4life — CHANGES_REQUESTED** [review]
Fantastic addtion! Great consistency across files and pipeline approach.  A couple of nits to pick; review or ignore as you wish!

File: static/marketplace/skills/dependabot-alerts-exporting/scripts/fetch-dependabot-alerts.mjs (hunk 1 above)
Comment (jordanpadams) [reply static/marketplace/skills/dependabot-alerts-exporting/scripts/fetch-dependabot-alerts.mjs]: Good catch — 403 has multiple meanings in the GitHub API (permission denied, repo access issues, and rate limits). This file has moved to PR #7 (feature/dependabot-skills). Fixed there: narrowed the retry logic to only treat 403 as a rate-limit when `x-ratelimit-remaining === '0'`, letting other 403s propagate as real errors.

File: static/marketplace/skills/dependabot-alerts-exporting/scripts/fetch-dependabot-alerts.mjs (hunk 2 above)
Comment (jordanpadams) [reply static/marketplace/skills/dependabot-alerts-exporting/scripts/fetch-dependabot-alerts.mjs]: Agreed — alphabetical sort gives HIGH, LOW, MEDIUM which is misleading. This file has moved to PR #7 (feature/dependabot-skills). Fixed there with an explicit `severityOrder = ['critical', 'high', 'medium', 'low']` array to guarantee the correct display order.
